### PR TITLE
add tests and ::new() for agent keys

### LINKED ISSUE #123: implement new for key and keys
child of #88 

currently there's no real interface for working with agent keys, a basic set of `new` fns can help

### PR DESCRIPTION
**Pull request by thedavidmeister**
fixes #123 

this fleshes out the agent keys just enough to make them useful while mocking things elsewhere...

### CODE REVIEW

File: core/src/agent/keys.rs [hunk 1]
```diff
@@ -1,9 +1,78 @@
...
 #[derive(Clone, Debug, PartialEq)]
 pub struct Keys {
     pub_key: Key,
     priv_key: Key,
     node_id: String,
 }
+
+impl Keys {
+
+    pub fn new(pub_key: &Key, priv_key: &Key, node_id: &str) -> Keys {
```
Comment (sphinxc0re) [inline core/src/agent/keys.rs]: I think it would be easier to use a `S: Into<String>`.

File: core/src/agent/keys.rs [hunk 2]
```diff
@@ -1,9 +1,78 @@
...
+        Keys {
+            pub_key: pub_key.clone(),
+            priv_key: priv_key.clone(),
+            node_id: String::from(node_id),
```
Comment (sphinxc0re) [inline core/src/agent/keys.rs]: Then, you could just do `node_id.into()` here.

File: core/src/agent/keys.rs [hunk 3]
```diff
@@ -1,9 +1,78 @@
...
+    use super::Keys;
+
+    /// generates a new key suitable for testing
+    pub fn test_key() -> Key {
+        Key::new()
+    }
+
+    /// generates a new node id suitable for testing
+    pub fn test_node_id() -> String {
+        String::from("test node id")
```
Comment (sphinxc0re) [inline core/src/agent/keys.rs]: you can just do:
```rust
"test node id".into()
```

File: core/src/agent/keys.rs (hunk 1 above)
Comment (thedavidmeister) [reply core/src/agent/keys.rs]: @sphinxc0re yar, it works, just seems like a lot of type juggling to support string arguments

**Pull request merged**